docs/architecture/tasks.md

```diff
@@ -0,0 +1,14 @@
+|          **Trigger**         |                   **Task**                  |
+|------------------------------|---------------------------------------------|
+| Fleet template updated       | Create new TemplateVersion                  |
```

Comment:
I wonder if we could have this in code instead. A table that maps the triggers into the tasks, where tasks repeat sometimes for different triggers.